src/sentry/incidents/logic.py

```diff
@@ -785,7 +785,7 @@ def update_alert_rule(
                 QueryAggregations(alert_rule.aggregation),
                 timedelta(minutes=alert_rule.time_window),
                 timedelta(minutes=DEFAULT_ALERT_RULE_RESOLUTION),
-                list(alert_rule.environment.all()),
+                alert_rule.environment.first(),
```

Comment:
Is first() going to apply and order and limit 1 and cause problems here?

Reply:
Hmm dang, I forgot about that. I'll avoid using this just to be safe.